python/GafferRenderManTest/RenderManShaderTest.py

```diff
@@ -289,5 +289,45 @@ def testDot( self ) :
 		self.assertTrue( script["shaderAssignment"]["shader"].getInput().isSame( script["dot"]["out"] ) )
 		self.assertTrue( script["shaderAssignment"]["shader"].source().isSame( script["shader"]["out"]["bxdf_out"] ) )
 
+	def testSplines( self ) :
+
+		shader = GafferRenderMan.RenderManShader()
+		shader.loadShader( "LamaIridescence" )
+
+		referencePlug = Gaffer.SplinefColor3fPlug( "targetHue",
+			defaultValue = Gaffer.SplineDefinitionfColor3f(
+				(
+					( 0, imath.Color3f( 1, 0.25, 0.25 ) ),
+					( 0.166, imath.Color3f( 1, 1, 0.25 ) ),
+					( 0.333, imath.Color3f( 0.25, 1, 0.25 ) ),
+					( 0.5, imath.Color3f( 0.25, 1, 1 ) ),
+					( 0.666, imath.Color3f( 0.25, 0.25, 1 ) ),
+					( 0.833, imath.Color3f( 1, 0.25, 1 ) ),
+					( 1, imath.Color3f( 1, 0.25, 0.25 ) ),
+				),
+				Gaffer.SplineDefinitionInterpolation( 1 )
+			)
+		)
+
+		self.assertEqual( shader['parameters']['targetHue'].getName(), referencePlug.getName() )
+		self.assertEqual( shader['parameters']['targetHue'].getFlags(), referencePlug.getFlags() )
+		self.assertEqual( shader['parameters']['targetHue'].direction(), referencePlug.direction() )
```

Comment:
Why make a reference plug, instead of just asserting direction and flags directly?